web/src/message_view_header.js

```diff
@@ -106,6 +109,30 @@ function build_message_view_header(filter) {
     }
 }
 
+// Add tooltip to stream name
+$("body").on("mouseenter", ".message-header-stream-settings-button", (e) => {
```

Comment:
Let's use the `delegate` addon function over here, you'll see multiple examples of such over at `tippyjs.js`. You can then use the `onShow(instance)` callback to extract the data and then display the rendered tooltip.

Reply:
Below is a good example of how it is handled.
```js
    message_list_tooltip(".message_edit_notice", {
        trigger: "mouseenter",
        delay: LONG_HOVER_DELAY,
        popperOptions: {
            modifiers: [
                {
                    name: "flip",
                    options: {
                        fallbackPlacements: "bottom",
                    },
                },
            ],
        },
        onShow(instance) {
            const $elem = $(instance.reference);
            const edited_notice_str = $elem.attr("data-tippy-content");
            instance.setContent(
                parse_html(
                    render_message_edit_notice_tooltip({
                        edited_notice_str,
                        realm_allow_edit_history: page_params.realm_allow_edit_history,
                    }),
                ),
            );
        },
        onHidden(instance) {
            instance.destroy();
        },
    });
```